### PR TITLE
Add systemd notifications

### PR DESCRIPTION
**Pull request by al1img**
Signed-off-by: Oleksandr Grytsov <al1img@gmail.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,13 +22,15 @@ message(STATUS "${PROJECT_NAME} Configuration:")
 message(STATUS "CMAKE_BUILD_TYPE              = ${CMAKE_BUILD_TYPE}")
 message(STATUS "CMAKE_INSTALL_PREFIX          = ${CMAKE_INSTALL_PREFIX}")
 message(STATUS)
-message(STATUS "WITH_DOC                      = ${WITH_DOC}")
+message(STATUS "WITH_SYSTEMD                  = ${WITH_SYSTEMD}")
```
Comment (lorc) [inline CMakeLists.txt]: Here you replace `WITH_DOC` with `WITH_SYSTEMD`...

File: CMakeLists.txt [hunk 2]
```diff
@@ -138,11 +145,13 @@ add_dependencies(${PROJECT_NAME} Version)
 # Docs
 ################################################################################
 
-find_package(Doxygen)
+if(WITH_DOC)
```
Comment (lorc) [inline CMakeLists.txt]: ... there you add `WITH_DOC`...

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -4,14 +4,20 @@
 
 include(FindPkgConfig)
 
-pkg_check_modules (DRM REQUIRED libdrm)
+if(WITH_SYSTEMD)
+	pkg_check_modules(SYSTEMD REQUIRED libsystemd)
+endif()
+
+if(WITH_DRM)
```
Comment (lorc) [inline src/CMakeLists.txt]: There you add check for `WITH_DRM`....

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -4,14 +4,20 @@
...
+endif()
 
 if(WITH_WAYLAND)
-	pkg_check_modules (WAYLAND REQUIRED wayland-client)
+	pkg_check_modules(WAYLAND REQUIRED wayland-client)
 endif()
 
 if(WITH_IVI_EXTENSION AND WITH_WAYLAND)
-	pkg_check_modules (IVI_APP ivi-application)
+	pkg_check_modules(IVI_APP ivi-application)
```
Comment (lorc) [inline src/CMakeLists.txt]: There you change formatting...

File: src/main.cpp [hunk 5]
```diff
@@ -78,8 +82,27 @@ string gLogFileName;
 /*******************************************************************************
  *
  ******************************************************************************/
+#ifdef WITH_SYSTEMD
+#define NOTIFY_SYSTEMD(...)														\
```
Comment (lorc) [inline src/main.cpp]: ... and only there you are really do something with systemd

File: src/main.cpp [hunk 6]
```diff
@@ -78,8 +82,27 @@ string gLogFileName;
...
+		LOG("Main", WARNING) << "systemd: NOTIFY_SOCKET is not set";			\
+	}																			\
+}
+
+#else
+#define NOTIFY_SYSTEMD(...)
+#endif
 
-void segmentationHandler(int sig)
+static void segmentationHandler(int sig)
```
Comment (lorc) [inline src/main.cpp]: That should belong to another patch.

File: src/main.cpp (hunk 5 above)
Comment (al1img) [reply src/main.cpp]: done
